### PR TITLE
Api: add default content type for all endpoints (except documentation…

### LINKED ISSUE #956: apirest should have a default content-type if not defined.
See https://github.com/glpi-project/glpi/commit/daacda142ed18dc2a635833765bafb6104c2a321#commitcomment-18961483

### PR DESCRIPTION
**Pull request by orthagh**
…); fix #956

### CODE REVIEW

**Comment by ddurieux** [comment]
It's ok for me for the modification.

Miss documentation of argument _$skip_check_content_type_ of the function.

So like I see, if _$skip_check_content_type_ = True the format is in HTML

That's a strange behavior for an API REST no?

**Comment by orthagh** [comment]
Good point for missing docblock.

Actually, asking for inline documentation (http://glpi/apirest.php/ url) with a browser don't send the content-type in the request headers.
In the previous commits fixing the travis tests and the absence of this header with versions <=5.6 of php and with php built-in server, i added this parameter to keep inline documentation working.

With the last modification, the parameter name is not very consistent.
I'll push a change here.

Thanks for the review.

**Comment by ddurieux** [comment]
It's nice now, I'm ok ;)

File: inc/apirest.class.php [hunk 1]
```diff
@@ -350,7 +350,7 @@ public function parseIncomingParams($skip_check_content_type = false) {
          $content_type = $_SERVER['HTTP_CONTENT_TYPE'];
       } else {
          if (!$skip_check_content_type) {
-            $this->returnError("No Content-Type header found", 400, "ERROR_CONTENT_TYPE_NOT_FOUND");
+            $content_type = "application/json";
```
Comment (newls) [inline inc/apirest.class.php]: Thanks @orthagh.
I think this is not only creating a security issue ^^
So it's ok.

**Comment by trasher** [comment]
Merged

**Pull request closed without merging**